src/relay/transforms/simplify_expr.cc

```diff
@@ -163,6 +141,69 @@ class SimplifyTranspose : public DFPatternRewrite {
     return x;
   }
 
+  String PermuteLayout(const String& layout, std::vector<int> axes) const {
+    std::string new_layout{};
```

Comment:
AFAICT it would require extra string copying and more code to use String here due to the in place updates. But if you feel strongly about this we can do it.

Reply:
Its not important, i will leave it to your call :slightly_smiling_face: